engine/src/main/java/org/terasology/config/flexible/FlexibleConfigManagerImpl.java

```diff
@@ -0,0 +1,151 @@
+                JsonReader jsonReader = new JsonReader(reader);
+
+                Map<SimpleUri, String> unusedSettings = Maps.newHashMap();
+
+                if (jsonReader.hasNext()) {
+                    jsonReader.beginObject();
+
+                    while (jsonReader.hasNext()) {
+                        SimpleUri id = new SimpleUri(jsonReader.nextName());
+                        String value = jsonReader.nextString();
+                        unusedSettings.put(id, value);
+                    }
+
+                    jsonReader.endObject();
+                }
+
+                flexibleConfig.setUnusedSettings(unusedSettings);
+            } catch (IOException e) {
+                logger.error("Failed to load config file {}, falling back on default config");
+            }
+        }
+    }
+
+    @Override
+    public void save() {
+        for (Entry<SimpleUri, FlexibleConfig> entry : flexibleConfigs.entrySet()) {
+            SimpleUri flexibleConfigUri = entry.getKey();
+            FlexibleConfig flexibleConfig = entry.getValue();
+
+            try (BufferedWriter writer = Files.newBufferedWriter(getPathForFlexibleConfig(flexibleConfigUri), TerasologyConstants.CHARSET)) {
+                JsonObject jsonObject = new JsonObject();
+
+                Map<SimpleUri, Setting> settingMap = flexibleConfig.getActiveSettings();
+                Map<SimpleUri, String> unusedSettings = flexibleConfig.getUnusedSettings();
+
+                for (Entry<SimpleUri, Setting> activeSetting : settingMap.entrySet()) {
+                    Setting setting = activeSetting.getValue();
+                    if (!setting.getValue().equals(setting.getDefaultValue())) {
+                        jsonObject.addProperty(activeSetting.getKey().toString(), setting.getValue().toString());
+                    }
+                }
+
+                // Add all the non-default settings that were not used in this session
+                if (unusedSettings != null) {
+                    for (Entry<SimpleUri, String> unusedSetting : unusedSettings.entrySet()) {
+                        jsonObject.addProperty(unusedSetting.getKey().toString(), unusedSetting.getValue());
+                    }
+                }
+
+                getGson().toJson(jsonObject, writer);
+            } catch (IOException e) {
+                logger.error("Failed to save config", e);
+            }
+        }
+    }
+
+    private void ensureDirectoryExists(Path filePath) {
+        try {
+            Files.createDirectories(filePath.getParent());
+        } catch (Exception e) {
+            throw new RuntimeException("Cannot create directory for FC!");
+        }
+    }
+
+    private Path getPathForFlexibleConfig(SimpleUri flexibleConfigUri) {
+        Path filePath = PathManager.getInstance().getHomePath().resolve("configs").resolve(flexibleConfigUri.getModuleName().toString()).resolve(flexibleConfigUri.getObjectName().toString() + ".cfg");
```

Comment:
Wowowowow. This line is too short. Can you make it any longer please? :smile: 

At least let's split it over multiple lines, i.e.

```
...getHomePath.resolve(blah)
              .resolve(blahblah)
              .resolve(blahblahblah)
```